lib/sqlalchemy/engine/util.py

```diff
@@ -97,6 +97,15 @@ def _trans_ctx_check(cls, subject: _TConsSubject) -> None:
         trans_context = subject._trans_context_manager
         if trans_context:
             if not trans_context._transaction_is_active():
+                transaction = getattr(subject, "transaction", None)
```

Comment:
@zzzeek is using `getattr` in this context acceptable, or would you prefer that the `_TConsSubject` protocol be extended to also require subjects passed to `_trans_ctx_check` have a `transaction` attribute?

Reply:
the protocol should be expanded and I would stick to underscored names in all cases